packages/diracx-web-components/src/components/shared/SearchBar/SearchField.tsx

```diff
@@ -0,0 +1,318 @@
+
+import type {
+  SearchBarTokenEquation,
+  SearchBarSuggestions,
+} from "../../../types";
+
+import { EquationAndTokenIndex } from "./Types";
+
+import {
+  getPreviousEquationAndToken,
+  handleEquationsVerification,
+  getTokenType,
+  convertListToString,
+} from "./Utils";
+
+interface SearchFieldProps {
+  inputValue: string;
+  setInputValue: React.Dispatch<React.SetStateAction<string>>;
+  inputRef: React.RefObject<HTMLInputElement>;
+  setTokenEquations: React.Dispatch<
+    React.SetStateAction<SearchBarTokenEquation[]>
+  >;
+  tokenEquations: SearchBarTokenEquation[];
+  suggestions: SearchBarSuggestions;
+  focusedTokenIndex: EquationAndTokenIndex | null;
+  setFocusedTokenIndex: React.Dispatch<
+    React.SetStateAction<EquationAndTokenIndex | null>
+  >;
+  allowKeyWordSearch?: boolean;
+}
+
+export default function SearchField({
+  inputValue,
+  setInputValue,
+  inputRef,
+  setTokenEquations,
+  tokenEquations,
+  suggestions,
+  focusedTokenIndex,
+  setFocusedTokenIndex,
+  allowKeyWordSearch = true,
+}: SearchFieldProps) {
+  const optionSelectedRef = useRef(false);
+  const { previousEquation, previousToken } = getPreviousEquationAndToken(
+    focusedTokenIndex,
+    tokenEquations,
+  );
+  const [placeholder, setPlaceholder] = useState<string>("");
+
+  useEffect(() => {
+    setPlaceholder(
+      previousToken
+        ? previousToken.type.startsWith("operator")
+          ? "Enter a value"
+          : previousToken.type.startsWith("category")
+            ? "Enter an operator"
+            : "Enter a category"
+        : "Enter a category",
+    );
+  }, [previousToken]);
+
+  /**
+   * Create a new token based on the input value and type.
+   * @param label The label for the new token.
+   * @param type The type of the token (e.g., "category", "custom", "operator", "value", ...).
+   */
+  function handleTokenCreation(label: string, type: string) {
+    if (!allowKeyWordSearch && type === "custom") return;
+
+    const formatedLabel = /\||,/.test(label)
+      ? label.split(/,|\|/).map((item) => item.trim())
+      : label.trim();
+
+    if (focusedTokenIndex) {
+      // If a token is focused, update the focused token
+      const updatedTokens = [...tokenEquations];
+      const equationIndex = focusedTokenIndex.equationIndex;
+      const tokenIndex = focusedTokenIndex.tokenIndex;
+
+      if (updatedTokens[equationIndex]) {
+        updatedTokens[equationIndex].items[tokenIndex] = {
+          label: formatedLabel,
+          type: type,
+          suggestions: !type.startsWith("category") ? suggestions : undefined,
+        };
+        updatedTokens[equationIndex].status =
+          type === "value" ? "valid" : "invalid";
+        handleEquationsVerification(updatedTokens, setTokenEquations);
+      }
+      setFocusedTokenIndex(null);
+    } else {
+      // If no token is focused, create a new token
+
+      if (previousEquation && previousEquation.status === "waiting") {
```

Comment:
You can also add an `enum` for the equation status? `waiting`, `valid`, `invalid`